# codonUsageToolkit

Toolkit to manipulate codon usage datasets

### About codonUsageToolkit

This toolkit allows you to:

1. *codonHeatmap*: build per-species/per-sequence codon usage heatmaps

2. *codonPCA*: perform PCA analysis of per-species/per-sequence codon usage RSCU values and plot scores and loadings plot

3. *protAbundanceVSrscu*: using a table of protein abundances for each gene, and a matched matrix of codon RSCU values for each of the genes, it builds plots depicting the correlation between RSCU and protein abundance, for each codon, and with individual plots for each amino acid

4. *ZcurveScore*: compute Zcurve-score metrics, and its dotproduct, as described in Guo et al., Nucl Acids Res 2003 (https://www.ncbi.nlm.nih.gov/pubmed/12626720). 

Detailed usage and input needed for each script can be found at the top of each individual script. 
Example data used in the scripts is found in the "data" folder included in this release.

### Citing this work:
If you find this work useful, please cite: Novoa EM, Jungreis IJ, Jaillon O, Kellis M. Elucidation of Codon Usage Signatures across the Domains of Life. bioRxiv 2018. https://www.biorxiv.org/node/128426.article-info

### License 
See LICENSE.md for details

### Contact
Please email us eva.novoa@crg.eu if you have any doubts/concerns/suggestions.
Thanks!
